File: src/test/java/main/QuetionsAndAnswers/ExportQandATest.java
package main.QuetionsAndAnswers;

import main.ApplicationsNavigationTest;
import main.PublicTenderPage;
import main.TendersPage;
import main.questionAndAnswers.ExportQandA;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.time.Instant;

import static org.testng.Assert.assertTrue;

public class ExportQandATest extends ApplicationsNavigationTest {

    @Test
    public void ExportQandAinWordAll() throws InterruptedException {
        TendersPage tendersPage = new TendersPage(driver);
        tendersPage.switchToBrowserFrame();

        PublicTenderPage publicTenderPage = new PublicTenderPage(driver);
        publicTenderPage.clickOnElement(publicTenderPage.getQuestionsAndAnswersTab());
        driver.switchTo().defaultContent();
        tendersPage.switchToNavigationFrame();
        ExportQandA exportQandA = new ExportQandA(driver);
        exportQandA.clickExportQA_button();
        tendersPage.switchToNewWindowTest();
        exportQandA.selectWordFormat();
        exportQandA.selectAll_QA();
        exportQandA.confirmOK();
        Thread.sleep(15000);
//        WebDriverWait driverWait = new WebDriverWait(driver, 16); // Зміна wait для очікування
//        driverWait.until(ExpectedConditions.presenceOfElementLocated(By.partialLinkText("docx")));
        assertTrue(exportQandA.link().isDisplayed());

    }
    @Test
    public void ExportQandAinExceldAll() throws InterruptedException {
        TendersPage tendersPage = new TendersPage(driver);
        tendersPage.switchToBrowserFrame();
        PublicTenderPage publicTenderPage = new PublicTenderPage(driver);
        publicTenderPage.clickOnElement(publicTenderPage.getQuestionsAndAnswersTab());
        driver.switchTo().defaultContent();
        tendersPage.switchToNavigationFrame();
        ExportQandA exportQandA = new ExportQandA(driver);
        exportQandA.clickExportQA_button();
        tendersPage.switchToNewWindowTest();
        exportQandA.selectExcelFormat();
        exportQandA.selectAll_QA();
        exportQandA.confirmOK();
        Thread.sleep(15000);
        assertTrue(exportQandA.link().isDisplayed());
    }
    @Test
    public void ExportQandAinExceldDrafts() throws InterruptedException {
        TendersPage tendersPage = new TendersPage(driver);
        tendersPage.switchToBrowserFrame();
        PublicTenderPage publicTenderPage = new PublicTenderPage(driver);
        publicTenderPage.clickOnElement(publicTenderPage.getQuestionsAndAnswersTab());
        driver.switchTo().defaultContent();
        tendersPage.switchToNavigationFrame();
        ExportQandA exportQandA = new ExportQandA(driver);
        exportQandA.clickExportQA_button();
        tendersPage.switchToNewWindowTest();
        exportQandA.selectExcelFormat();
        exportQandA.selectDrafts();
        exportQandA.confirmOK();
        Thread.sleep(15000);
        assertTrue(exportQandA.link().isDisplayed());
    }  @Test
    public void ExportQandAinExceldUnanswered() throws InterruptedException {
        TendersPage tendersPage = new TendersPage(driver);
        tendersPage.switchToBrowserFrame();
        PublicTenderPage publicTenderPage = new PublicTenderPage(driver);
        publicTenderPage.clickOnElement(publicTenderPage.getQuestionsAndAnswersTab());
        driver.switchTo().defaultContent();
        tendersPage.switchToNavigationFrame();
        ExportQandA exportQandA = new ExportQandA(driver);
        exportQandA.clickExportQA_button();
        tendersPage.switchToNewWindowTest();
        exportQandA.selectExcelFormat();
        exportQandA.selectUnanswered();
        exportQandA.confirmOK();
        Thread.sleep(15000);
        assertTrue(exportQandA.link().isDisplayed());
    } @Test
    public void ExportQandAinExceldPublished() throws InterruptedException {
        TendersPage tendersPage = new TendersPage(driver);
        tendersPage.switchToBrowserFrame();
        PublicTenderPage publicTenderPage = new PublicTenderPage(driver);
        publicTenderPage.clickOnElement(publicTenderPage.getQuestionsAndAnswersTab());
        driver.switchTo().defaultContent();
        tendersPage.switchToNavigationFrame();
        ExportQandA exportQandA = new ExportQandA(driver);
        exportQandA.clickExportQA_button();
        tendersPage.switchToNewWindowTest();
        exportQandA.selectExcelFormat();
        exportQandA.selectPublished();
        exportQandA.confirmOK();
        Thread.sleep(15000);
        assertTrue(exportQandA.link().isDisplayed());
    }
}

File: src/test/java/main/QuetionsAndAnswers/ExportQandATest.kt
package main.QuetionsAndAnswers

import junit.framework.Assert.assertTrue
import main.ApplicationsNavigationTest
import main.PublicTenderPage
import main.TendersPage
import main.questionAndAnswers.ExportQandA
import org.openqa.selenium.support.ui.ExpectedConditions
import org.openqa.selenium.support.ui.WebDriverWait
import org.testng.annotations.BeforeMethod
import org.testng.annotations.Test

class ExportQandATest : ApplicationsNavigationTest() {
    private lateinit var exportQandA: ExportQandA

    @BeforeMethod
    fun setUpTest() {
        val tendersPage = TendersPage(driver)
        tendersPage.switchToBrowserFrame()
        val publicTenderPage = PublicTenderPage(driver)

        publicTenderPage.questionsAndAnswersTab?.click()
        driver.switchTo().defaultContent()
        tendersPage.switchToNavigationFrame()
        exportQandA = ExportQandA(driver)
        driver.findElement(exportQandA.exportQuestionButton).click()
        tendersPage.switchToNewWindowTest() }

    @Test
    fun ExportQandAinWordAll() {
        driver.findElement(exportQandA.wordFormatOption).click()
        exportQandA.selectAll_QA()
        driver.findElement(exportQandA.okButton).click()
        val linkElement = WebDriverWait(driver, 45).until(ExpectedConditions.visibilityOf(exportQandA.linkQAElement))
        assertTrue(linkElement.isDisplayed) }

    @Test
    fun ExportQandAinExceldAll() {
        driver.findElement(exportQandA.excelFormatOption).click()
        exportQandA.selectAll_QA()
        driver.findElement(exportQandA.okButton).click()
        val linkElement = WebDriverWait(driver, 45).until(ExpectedConditions.visibilityOf(exportQandA.linkQAElement))
        assertTrue(linkElement.isDisplayed) }

    @Test
    fun ExportQandAinExceldDrafts() {
        driver.findElement(exportQandA.excelFormatOption).click()
        exportQandA.selectDrafts()
        driver.findElement(exportQandA.okButton).click()
        val linkElement = WebDriverWait(driver, 45).until(ExpectedConditions.visibilityOf(exportQandA.linkQAElement))
        assertTrue(linkElement.isDisplayed) }
    @Test
    fun ExportQandAinExceldUnanswered() {
        driver.findElement(exportQandA.excelFormatOption).click()
        exportQandA.selectUnanswered()
        driver.findElement(exportQandA.okButton).click()
        val linkElement = WebDriverWait(driver, 45).until(ExpectedConditions.visibilityOf(exportQandA.linkQAElement))
        assertTrue(linkElement.isDisplayed) }

    @Test
    fun ExportQandAinExceldPublished() {
        driver.findElement(exportQandA.excelFormatOption).click()
        exportQandA.selectPublished()
        driver.findElement(exportQandA.okButton).click()
        val linkElement = WebDriverWait(driver, 45).until(ExpectedConditions.visibilityOf(exportQandA.linkQAElement))
        assertTrue(linkElement.isDisplayed) }
}